python/mlcroissant/mlcroissant/_src/operation_graph/operations/parse_json.py

```diff
@@ -7,10 +7,15 @@
 from mlcroissant._src.structure_graph.nodes.field import Field
 
 
-def parse_json_content(json: Json, fields: tuple[Field, ...]) -> pd.DataFrame:
+def parse_json_content(
+    json: Json, fields: tuple[Field, ...], record_set: str | None = None
+) -> pd.DataFrame:
     """Parsed all JSONs defined in the fields of RecordSet and outputs a pd.DF."""
     series = {}
     for field in fields:
+        # Only select fields relevant to the requested record_set (if given).
+        if record_set and not field.uid.startswith(record_set):
```

Comment:
Here can you filter at the field level (`fields`) rather than adding another parameter (`record_set`)?

Reply:
Good point, changed. WDYT?